source/upgrade/upgrade_notes.rst

```diff
@@ -4,6 +4,19 @@
 Upgrade notes
 *************
 
+19.17
+=====
+
+* ``dahdi`` is not longer a dependency on Wazo. Upgraded engines will include all dependencies to
+  have a working dahdi installation. If you do not want dahdi installed on your system execute the
+  following steps::
+
+    rm -f /etc/wazo-confgend/conf.d/50_enable_chan_dahdi.yml
+    systemctl restart wazo-confgend
+    # THIS WILL RESTART ASTERISK
+    apt remove wazo-asterisk-extra-modules
+
```

Comment:
Since 99% of upgrade will do this procedure, should we invert it? Remove dahdi by default and if needed, reinstall it

Reply:
I must say that I have a preference for not breaking the install but not a strong preference.

@mwolff44 do you have an option about this change?

Do you prefer to
1. keep dahdi enabled on upgrades with instructions to disable it in the upgrade notes
2. disable dahdi on upgrades with instructions to enable it in the upgrade notes

Reply:
Another impact about this change is maintainability

What/why are advantaged to remove dahdi installed by default
* Not used by default anymore
* Break a lot of upgrades
  * We never put time on it

If we keep installed by default, we lose the gain that we would have done by removing it for upgrades.
Even if we still need to support upgrade for those who have the package, the impact should be minimal if the package is installed only for those who need it.

I have not found, in my memory, such decision with the same situation (upgrading a component installed before, but not anymore by default and still supporting)

* I can think of our wazo-*-client, when we need to use it anymore, the autoremove just remove the package and the user must reinstall it manually (I know, it is not 100% the same thing, the user can install manually before)
* With MongooseIM, we do not use it and we remove it. But again, it is not the same thing.

Since I don't know what the real number of people using it explicitly,  it's hard to choose wisely.
Maybe @sduthil have others arguments

Reply:
To be confirmed with @sboily or @mwolff44 but I think the goal of removing the Dahdi dependency is to be able to install Wazo easily on providers where installing the Dahdi Kernel modules is not possible. Not to remove the burden of maintaining it.

Reply:
+1

Reply:
* If we remove dahdi, we will break the old meetme conference that haven't been migrated from 18.03 yet. So we need to decide how to manage this migration (only an upgrade note + script or a fully automated migration)
* If `/etc/asterisk/dahdi-channels.conf` is present on a machine, we can safely assume that DAHDI is used. If not, then we can remove the package. We should still write something in the upgrade note for the admins saying "check that DAHDI is still there if you need it. If not, install it like this: ..."